src/status_im/contexts/profile/contact/share/view.cljs

```diff
@@ -0,0 +1,53 @@
+(ns status-im.contexts.profile.contact.share.view
+  (:require [legacy.status-im.ui.components.list-selection :as list-selection]
+            [quo.core :as quo]
+            [react-native.core :as rn]
+            [react-native.safe-area :as safe-area]
+            [status-im.common.qr-codes.view :as qr-codes]
+            [status-im.contexts.profile.contact.share.style :as style]
+            [status-im.contexts.profile.utils :as profile.utils]
+            [utils.address :as address]
+            [utils.i18n :as i18n]
+            [utils.re-frame :as rf]))
+
+(defn view
+  []
+  (let [{:keys [universal-profile-url customization-color]
+         :as   profile} (rf/sub [:contacts/current-contact])
+        abbreviated-url (rn/use-memo (fn []
+                                       (address/get-abbreviated-profile-url universal-profile-url))
+                                     [universal-profile-url])
+        profile-picture (rn/use-memo #(profile.utils/photo profile) [])
```

Comment:
should we not add `profile` as a dependency of the memo function?

Reply:
I believe it should be, otherwise I am guessing that it won't be updated if the profile picture changes?

Reply:
yes, sorry i missed that. we may need something like exhaustive deps lint rule (`eslint-plugin-react-hooks` lib) or we will miss dependency array easily

Reply:
One thing that might come up is that the dependency array may expect only primitive values (I think). I recall reading somewhere that our helper functions (`rn/use-memo`) don't handle converting and comparing maps or object types by structural equality. I may be incorrect about that, but it may be worth double checking.

Reply:
Actually it might be fine to pass whole CLJS values 🤔 

For context:
https://github.com/status-im/status-mobile/blob/b3bebe18595c8e8d24b474c48e5aa4b1ac998bfd/src/react_native/core.cljs#L196-L198

https://github.com/status-im/status-mobile/blob/b3bebe18595c8e8d24b474c48e5aa4b1ac998bfd/src/react_native/core.cljs#L156-L170

Reply:
Yes, we added support for comparing clojure datatypes, so in theory it supports complex objects, since it will be clojure-comparison them, but probably still best to pass primitives whenever we can, just so that usage is consistent with plain react and people don't get confused